AK/Optional.h

```diff
@@ -58,13 +62,22 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
 public:
     using ValueType = T;
 
-    ALWAYS_INLINE Optional() = default;
+    // FIXME: Not sure why this constructor is needed,
+    //        as it essentially does the same thing as the default constructor,
+    //        as the union already has an initialized member,
+    //        but for some reason compilers assert having default constructor
+    //        on the value type
+    ALWAYS_INLINE constexpr Optional()
+    requires(!IsConstructible<T>)
+    {
+    }
+    ALWAYS_INLINE constexpr Optional() = default;
```

Comment:
Empty line between this and the previous decl please.

Reply:
Although this was intentional, ok